File: Assets/Script/EventButton.cs
using UnityEngine;
using UnityEngine.UI;
public class EventButton : MonoBehaviour {
	public string eventName;

	public void Awake () {
		GetComponent<Button>().onClick.AddListener(OnClick);
	}
	public void OnClick() {
		if(null != eventName) {
			EventManager.Instance.CastEvent(this, eventName, null);
		}
	}
}

File: Assets/Script/EventButton.js
#pragma strict
import UnityEngine.UI;
var eventName: String;
private var eventCenter: EventCenter;

function Awake () {
	eventCenter = GameObject.FindWithTag('EventCenter').GetComponent(EventCenter);
	GetComponent(Button).onClick.AddListener(OnClick);
}
function OnClick() {
	if(null != eventName) {
		eventCenter.CastEvent(this, eventName, null);
	}
}
